src/easystar.js

```diff
@@ -286,6 +290,23 @@ EasyStar.js = function() {
             instance.startY, null, STRAIGHT_COST));
 
         instances.push(instance);
+        return instance.id;
+    };
+
+    /**
+     * Cancel a path calculation.
+     *
+     * @param {Number} id The ID of the path being calculated.
+     *
+     **/
+    this.cancelPath = function(id) {
+        for (var i = 0; i < instances.length; i ++) {
+            var instance = instances[i];
```

Comment:
Changing it to a map is possible, but requires heavier changes.  To get the performance benefit, the map (from instance ID to instance) would need to be the authoritative list of instances, not the current instances array.  But the instances array is used as the "queue" in `calculate()`.  To keep that queue FIFO, we'd want an array of IDs so that order is preserved.  So basically:

- `instances` becomes a map of ID to instance
- add new `instanceQueue` array of IDs

I'll put in a commit for that so you can decide if you think it's worth it.

true/false: No problem.